### PR TITLE
Add rule S6565 (`prefer-return-this-type`): Prefer return type `this` in fluent interfaces

### LINKED ISSUE #3815: Add rule for `prefer-return-this-type`
https://typescript-eslint.io/rules/prefer-return-this-type

 - Covered?
   - Not covered
 - Makes sense?
   - Enforce that the type `this` is used when only `this` is returned
 - No options, quickfix available

### PR DESCRIPTION
**Pull request by francoismora**
This PR adds the `typescript-eslint` rule `prefer-return-this-type` which includes a quick-fix.

Fixes #3815 
RSPEC: SonarSource/RSPEC#1715

### CODE REVIEW

File: tests/linting/eslint/rules/prefer-return-this-type.test.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { rules as typescriptESLintRules } from '@typescript-eslint/eslint-plugin';
+import { TypeScriptRuleTester } from '../../../tools';
+
+const ruleTester = new TypeScriptRuleTester();
+const rule = typescriptESLintRules['prefer-return-this-type'];
+
+ruleTester.run(`Rule should provide suggestion with proper message`, rule, {
```
Comment (ilia-kebets-sonarsource) [inline tests/linting/eslint/rules/prefer-return-this-type.test.ts:26]:
```suggestion
ruleTester.run(`Rule should provide a quickfix suggestion with the proper message`, rule, {
```

File: tests/linting/eslint/rules/prefer-return-this-type.test.ts
Comment (ilia-kebets-sonarsource) [inline tests/linting/eslint/rules/prefer-return-this-type.test.ts:1]: why not use the comment-based framework?

**Review by ilia-kebets-sonarsource — APPROVED** [review]
See comments otherwise LGTM!

File: tests/linting/eslint/rules/prefer-return-this-type.test.ts
Comment (francoismora) [reply tests/linting/eslint/rules/prefer-return-this-type.test.ts:1]: I prefer the syntax of rule testers to validate multi-line quick fixes.

**Pull request merged**